core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -146,7 +146,7 @@ public Set<Integer> struct(Types.StructType struct, Iterable<Set<Integer>> field
 
             // visit children to add more ids
             Iterator<Set<Integer>> iter = fieldResults.iterator();
-            while (shouldContinue() && iter.hasNext()) {
+            while (iter.hasNext()) {
               iter.next();
```

Comment:
I'm very confused about this, what does iter.next() do here? This looks like a noop? (although I didn't understand what it was doing before either

Reply:
thanks @RussellSpitzer , also added one more test to highlight the difference in https://github.com/apache/iceberg/pull/13785/files#r2279871102